kolibri/core/assets/src/api_resource.js

```diff
@@ -27,36 +50,156 @@ class Model {
+   */
+  save(attrs) {
+    const promise = new Promise((resolve, reject) => {
+      Promise.all(this.promises).then(() => {
+        let payload = {};
+        if (this.synced) {
+          // Model is synced with the server, so we can do dirty checking.
+          Object.keys(attrs).forEach((key) => {
+            if (attrs[key] !== this.attributes[key]) {
+              payload[key] = attrs[key];
+            }
+          });
+        } else {
+          payload = attrs;
+        }
+        if (!Object.keys(payload).length) {
+          // Nothing to save, so just resolve the promise now.
+          resolve(this.attributes);
+        } else {
+          this.synced = false;
+          let url;
+          let clientObj;
+          if (this.id) {
+            // If this Model has an id, then can do a PATCH against the Model
+            url = this.url;
+            clientObj = { path: url, method: 'PATCH' };
+          } else {
+            // Otherwise, must POST to the Collection endpoint to create the Model
+            url = this.resource.collectionUrl();
+            clientObj = { path: url, entity: payload,
+              headers: { 'Content-Type': 'application/json' } };
+          }
+          // Do a save on the URL.
+          client(clientObj).then((response) => {
+            const oldId = this.id;
+            // Set the retrieved Object onto the Model instance.
+            this.set(response.entity);
+            // if the model did not used to have an id and now does, add it to the cache.
+            if (!oldId && this.id) {
+              this.resource.addModel(this);
+            }
+            // Flag that the Model has been fetched.
+            this.synced = true;
+            // Resolve the promise with the Model.
+            resolve(this);
+            // Clean up the reference to this promise
+            this.promises.splice(this.promises.indexOf(promise), 1);
+          }, (response) => {
+            logging.error('An error occurred', response);
+            reject(response);
+            // Clean up the reference to this promise
+            this.promises.splice(this.promises.indexOf(promise), 1);
+          });
+        }
+      },
+      (reason) => {
+        reject(reason);
       });
     });
+    this.promises.push(promise);
+    return promise;
+  }
+
+  delete(id) {
+    const promise = new Promise((resolve, reject) => {
+      Promise.all(this.promises).then(() => {
+        if (!this.id) {
+          // Nothing to delete, so just resolve the promise now.
+          resolve(this.attributes);
+        } else {
+          // Otherwise, DELETE the Model
+          const clientObj = { path: this.url, method: 'DELETE',
+            headers: { 'Content-Type': 'application/json',
+              'X-CSRFToken': this.getCookie('csrftoken') } };
+          // Do a save on the URL.
+          client(clientObj).then((response) => {
+            // delete this instance
+            // this.prototype.delete();
```

Comment:
@rtibbles not sure how to delete the Model instance inside api_resource.js Model delete(id) method